enhancements/sig-policy/99-policy-placement-strategy/README.md

```diff
@@ -0,0 +1,269 @@
+    Default to `All` if a strategy is not provided.
+  - Update the `governance-policy-propagator` controller to add a `open-cluster-management.io/root-policy-timestamp`
+    annotation to the replicated policy for each cluster indicating when the replicated policy was last applied.
+  - Store a `lastEvaluated` in the status of the replicated policies so that the `RolloutStatus` can be determined
+    inside of `ClusterRolloutStatusFunc`. This should only be updated if the status changes (current behavior), or the
+    generation changes and/or the root policy timestamp is less than the `lastEvaluated` timestamp.
+  - Update the `ClusterRolloutStatusFunc`, (the `GetClusterRolloutStatus` function from Phase 2) to set up logic to work
+    with the `Placement` library to provide the information it needs to continue only once the previous group is
+    compliant (or after a configurable timeout):
+
+    | Status        | Description                                                                                                                                          |
+    | ------------- | ---------------------------------------------------------------------------------------------------------------------------------------------------- |
+    | `ToApply`     | No generation set                                                                                                                                    |
+    | `Progressing` | Current generation and `lastEvaluated` has not been updated                                                                                          |
+    | `Succeed`     | Compliant and current generation                                                                                                                     |
+    | `Failed`      | NonCompliant and current generation                                                                                                                  |
+    | `TimeOut`     | Current generation, `lastEvaluated` is less than `root-policy-timestamp`, </br>and time has passed beyond timeout specified in the `RolloutStrategy` |
+    | `Skip`        | (unused)                                                                                                                                             |
+
+  - (Stretch) Update the `Policy` CRD to contain a `RolloutStatus`. This will be used to provide an aggregated rollout
+    status for the root policy of `Progressing` (this would include `ToApply`), `Succeed`, or `Failed` (this would
+    include `TimeOut`).
+
+- **Snippet**
+
+  ```golang
+  type PlacementBinding struct {
+    metav1.TypeMeta   `json:",inline"`
+    metav1.ObjectMeta `json:"metadata,omitempty"`
+    // +kubebuilder:validation:Optional
+    BindingOverrides BindingOverrides `json:"bindingOverrides,omitempty"`
+    // +kubebuilder:validation:Optional
+    RolloutStrategy clusterv1alpha1.RolloutStrategy `json:"rolloutStrategy,omitempty"` // <-- New field
```

Comment:
Good questions. I need to figure out how selective policy enforcement works, but I think they could work together, with the propagator filtering out clusters from the clusters returned from the Placement library.

The `additionalPrinterColumns` would be a good addition to the CRD!

Reply:
@dhaiducek I worry about the rollout strategy being defined on the PlacementBinding instead of the Policy because you can have multiple PlacementBindings per Policy. If multiple set rolloutStrategy, which one wins? Could a subset of clusters rollout `Progressive` and others rollout `ProgressivePerGroup`?

Also, the PlacementBinding can be used for multiple policies and it'd be annoying to have to use a separate PlacementBinding just because your policies have different rollout strategies.

Lastly, I think the UX around PlacementBinding is challenging. I find it much more intuitive to configure how my policy is deployed by setting it on the policy.

Reply:
In regards to selective policy enforcement, I personally think it should be ignored or an error returned when a rollout strategy is also defined.

Reply:
The multiple placement bindings question also seems to apply to the order/timing of when the enforce is done. If two bindings for the same policy have overlapping sets of clusters included both will (as I understand it) calculate decisiongroups independently. Any given cluster may be in group 1 of binding A and group 100 of binding B. Will it get enforced twice? If the max concurrency is 10, do you end up with 11 clusters being enforced due to the different groupings? etc.